scripts/manage_pes

```diff
@@ -148,456 +89,355 @@ USAGE, OPTIONS, AND EXAMPLES
+
+            matchnodes.append(gridnode)
+        return matchnodes
+
+    def checkFile(self, pesxml):
+        gridnodes = pesxml.get_nodes(nodename="grid")
+        if len(gridnodes) == 0:
+            logger.critical(' xml file %s has no <grid> entry\n', self.filename)
+            sys.exit(1)
+        if len(gridnodes) > 1:
+            logger.critical(' xml file %s has multiple <grid> entries.'
+                            ' Not supported yet\n', self.filename)
+            sys.exit(1)
+
+        for gridnode in gridnodes:
+            gridname = gridnode.get("name")
+            if gridname is None:
+                logger.critical(' xml file %s has no <grid name="">'
+                                ' attribute \n', self.filename)
+                sys.exit(1)
+
+            machnode = gridnode.find("mach")
+            if machnode is None:
+                logger.critical(' xml file %s has no <grid><mach> entry\n',
+                                self.filename)
+                sys.exit(1)
+
+            machname = machnode.get("name")
+            if machname is None:
+                logger.critical(' xml file %s has no <mach name=""> '
+                                'attribute\n', self.filename)
+                sys.exit(1)
+
+            pesnode = machnode.find("pes")
+            if pesnode is None:
+                logger.critical(' xml file %s has no <grid><mach><pes> '
+                                'entry\n', self.filename)
+                sys.exit(1)
+
+            pesizename = pesnode.get("pesize")
+            if pesizename is None:
+                logger.critical(' xml file %s has no <pes pesize=""> '
+                                'attribute\n', self.filename)
+                sys.exit(1)
+
+            compsetname = pesnode.get("compset")
+            if compsetname is None:
+                logger.critical(' xml file %s has no <pes compsetname=""> '
+                                'attribute\n', self.filename)
+                sys.exit(1)
+
+            logger.debug("grid: %s\n\tmach: %s\n\tpesize: %s\n\tcompset: %s\n",
+                         gridname, machname, pesizename, compsetname)
+            return (gridname, machname, pesizename, compsetname)
+
+    def addPes(self):
+        """
+        Add a new pe-layout for a target machine and grid.
+        """
+        filesxml = Files()
+        compsetfile = None
+        compsetfile = filesxml.get_value("COMPSETS_SPEC_FILE",
+                                    attribute={"component":self.setby})
+        pesxmlfile = filesxml.get_value("PES_SPEC_FILE",
+                                    attribute={"component":self.setby})
+        # Try to collapse any '../' in filenames
+        # cosmetic only, don't worry if it breaks
+        try:
+            import posixpath
+            p1 = posixpath.normpath(compsetfile)
+            p2 = posixpath.normpath(pesxmlfile)
+            compsetfile = p1
+            pesxmlfile = p2
+        except Exception, e:
```

Comment:
Why catch such a general exception? This can hide syntax mistakes. There's a special exception type for import failures.